### PR TITLE
Fix ActiveSupport::TaggedLogging silence usage

### PR DESCRIPTION
**Pull request by luismiramirez**
When using ActiveSupport::TaggedLogging without the broadcast feature, any call to `silence` fails.

This patch fixes that behaviour by adding the `silence` method to our own logger. It acts as the one defined when using `ActiveSupport`'s broadcast feature. It just calls the passed block and let's the execution to continue.

This has to be done in order to add full support to `ActiveSupport::TaggedLogging`.

Part of: #948 

---

[Maintainer stating that tagged logging is only intended to work with loggers that respond to `silence`.](https://github.com/rails/sprockets-rails/issues/366#issuecomment-229812270)

### CODE REVIEW

File: .changesets/fix-a-bug-when-using-activesupport--taggedlogging.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+bump: "patch"
+type: "fix"
+---
+
+Fix a bug when using ActiveSupport::TaggedLogging
```
Comment (tombruijn) [inline .changesets/fix-a-bug-when-using-activesupport--taggedlogging.md:6]:
```suggestion
Fix a bug when using ActiveSupport::TaggedLogging calling the `silence` method.
```

File: spec/lib/appsignal/logger_spec.rb [hunk 2]
```diff
@@ -67,6 +67,18 @@
...
 
+  describe "#silence" do
+    it "calls the given block" do
+      num = 1
+
+      logger.silence do
+        num += 1
+      end
+
+      expect(num).to eq(2)
```
Comment (tombruijn) [inline spec/lib/appsignal/logger_spec.rb:78]:
```suggestion
      expect(num).to eq(2)
      expect(Appsignal::Extension).to_not receive(:log)
```

Could help to make sure we don't log anything through this method.

**Review by tombruijn — APPROVED** [review]
If build green 👍

**Review by unflxw — APPROVED** [review]

File: lib/appsignal/logger.rb [hunk 3]
```diff
@@ -114,6 +114,19 @@ def fatal(message = nil, attributes = {})
...
 
+    # When using ActiveSupport::TaggedLogging without the broadcast feature,
+    # the passed logger is required to respond to the `silence` method.
+    # In our case it behaves as the broadcast feature of the Rails logger, but
+    # we don't have to check if the parent logger has the `silence` method defined
+    # as our logger directly inherits from Ruby base logger.
+    #
+    # Links:
+    # https://github.com/rails/rails/blob/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/activesupport/lib/active_support/logger.rb#L60-L76
+    # https://github.com/rails/rails/blob/main/activesupport/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/active_support/logger_silence.rb
```
Comment (tombruijn) [inline lib/appsignal/logger.rb:125]:
```suggestion
    # rubocop:disable Layout/LineLength
    # https://github.com/rails/rails/blob/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/activesupport/lib/active_support/logger.rb#L60-L76
    # https://github.com/rails/rails/blob/main/activesupport/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/active_support/logger_silence.rb
    # rubocop:enable Layout/LineLength
```
These lines probably fail RuboCop when merged after merging #952.

File: lib/appsignal/logger.rb [hunk 4]
```diff
@@ -114,6 +114,19 @@ def fatal(message = nil, attributes = {})
...
+    def silence(_severity = ERROR, &block)
+      block.call
```
Comment (tombruijn) [inline lib/appsignal/logger.rb:127]:
```suggestion
    def silence(_severity = ERROR)
      yield
```
Can also use `yield` here directly.

File: lib/appsignal/logger.rb [hunk 5]
```diff
@@ -126,6 +126,19 @@ def fatal(message = nil, attributes = {})
...
+    # When using ActiveSupport::TaggedLogging without the broadcast feature,
+    # the passed logger is required to respond to the `silence` method.
+    # In our case it behaves as the broadcast feature of the Rails logger, but
+    # we don't have to check if the parent logger has the `silence` method defined
+    # as our logger directly inherits from Ruby base logger.
+    #
+    # Links:
+    # https://github.com/rails/rails/blob/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/activesupport/lib/active_support/logger.rb#L60-L76
+    # https://github.com/rails/rails/blob/main/activesupport/e11ebc04cfbe41c06cdfb70ee5a9fdbbd98bb263/active_support/logger_silence.rb
+    def silence(_severity = ERROR, &block)
```
[the remark this answers is not in the document]
Comment (luismiramirez) [reply lib/appsignal/logger.rb:138]: Yep, I noticed. I'm not sure if we should go back to using `block.call` this method's signature is different from the original

[the remark this answers is not in the document]
Comment (tombruijn) [reply lib/appsignal/logger.rb:138]: Whether you use yield or block doesn't matter for the method signature. It's a Ruby syntax thing only.

**Pull request merged**